<?xml version="1.0" ?>

<launch>
    <arg name="robot_type" default="$(env ROBOT_TYPE)" doc="Robot type: [a1, aliengo, go1, laikago]"/>
    <arg name="cheater" default="false"/>

    <param name="/rosconsole/configure" value="true"/>
    <param name="/rosconsole/logger_level" value="ros.tf2:ERROR"/>

    <!-- make the files into global parameters -->
    <param name="urdfFile" value="/tmp/legged_control/$(arg robot_type).urdf"/>
    <param name="taskFile" value="$(find legged_controllers)/config/$(arg robot_type)/task.info"/>
    <param name="referenceFile" value="$(find legged_controllers)/config/$(arg robot_type)/reference.info"/>
    <param name="gaitCommandFile" value="$(find legged_controllers)/config/$(arg robot_type)/gait.info"/>
    <rosparam file="$(find legged_controllers)/config/controllers.yaml" command="load"/>

    <node if="$(arg cheater)" name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="load
          controllers/joint_state_controller
          controllers/legged_controller
          controllers/legged_cheater_controller"/>

    <node unless="$(arg cheater)" name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="load
          controllers/joint_state_controller
          controllers/legged_controller"/>

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_gait_command" name="legged_robot_gait_command"
          output="screen" args="__log_level:=ERROR"/>

    <node pkg="legged_controllers" type="legged_target_trajectories_publisher" name="legged_robot_target"
          output="screen" args="__log_level:=ERROR"/>
          
    <node name="kp_kd_optimizer" pkg="legged_controllers" type="kp_kd_optimizer.py" output="screen"/>

    <!-- <node name="node_01" pkg="legged_controllers" type="114514" output="screen"/> -->
    <!-- <node name="kp_kd_optimizer" pkg="legged_controllers" type="LeggedRobot.py" output="screen"/> -->

</launch>
